src/main/java/org/apache/chemistry/opencmis/inmemory/storedobj/impl/ObjectStoreImpl.java

```diff
@@ -825,7 +851,10 @@ public void move(StoredObject so, Folder oldParent, Folder newParent,
 				addParentIntern(fi, newParent);
 				removeParentIntern(fi, oldParent);
 			} else if (so instanceof FolderImpl) {
+				// remove MapEntry with old path
+				fStoredObjectMap.remove(((FolderImpl) so).getPath());
 				((FolderImpl) so).setParentId(newParent.getId());
+				fStoredObjectMap.put(((FolderImpl) so).getPath(), so);
```

Comment:
If i understand well, you store full "cmisMetadata" object for the "path" too like "objectId".

In that case, the memory will increase a lot.

Maybe it will be better to store juste the "objectId" for the value corresponding to a path.

What do you think ?

Reply:
I share the StoredObject's reference into two hashMap entries. One for path and the other for objectId.
The memory will be increasing lightly but not as much as you thougth ;) The StoredObject will be exist only one-time into the Stack. But with two entries in the Heap.

Reply:
I understand two different key but the same pointer for the value. Ok.